### PR TITLE
added emails digest edge cases

### PR DESCRIPTION
**Pull request by BilalG1**
----

> [!IMPORTANT]
> Handle multiple tenant owner emails in failed email digests and add tests for edge cases.
> 
>   - **Behavior**:
>     - Update `getFailedEmailsByTenancy` in `crud.tsx` to handle multiple `tenantOwnerEmails`.
>     - Modify `POST` handler in `route.ts` to send emails to all `tenantOwnerEmails`.
>     - Skip sending digest if no `tenantOwnerEmails` are present.
>   - **Tests**:
>     - Add tests in `failed-emails-digest.test.ts` for scenarios: no primary email, no owner, deleted project, and multiple owners.
>     - Ensure digest is not sent if no owner or project is deleted.
>     - Verify digest is sent to all owners if multiple exist.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=stack-auth%2Fstack-auth&utm_source=github&utm_medium=referral)<sup> for 944f05c19283882dd05deec573388b5c71b0a502. You can [customize](https://app.ellipsis.dev/stack-auth/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: apps/backend/src/app/api/latest/internal/failed-emails-digest/route.ts [hunk 1]
```diff
@@ -50,28 +50,34 @@ export const POST = createSmartRouteHandler({
...
-          });
+          for (const tenantOwnerEmail of failedEmailsBatch.tenantOwnerEmails) {
+            await sendEmail({
+              tenancyId: internalTenancy.id,
+              emailConfig,
+              to: tenantOwnerEmail,
+              subject: "Failed emails digest",
+              html: emailHtml,
+            });
+          }
```
Comment (N2D4) [inline apps/backend/src/app/api/latest/internal/failed-emails-digest/route.ts]: `to` actually takes a string array, so you don't have to call the function multiple times

File: apps/e2e/tests/backend/endpoints/api/v1/internal/failed-emails-digest.test.ts [hunk 2]
```diff
@@ -65,17 +65,6 @@ describe("with valid credentials", () => {
     const adminAccessToken = backendContext.value.userAuth?.accessToken;
```
Comment (N2D4) [inline apps/e2e/tests/backend/endpoints/api/v1/internal/failed-emails-digest.test.ts]: remove unused variables when you can

```suggestion
```

File: apps/e2e/tests/backend/endpoints/api/v1/internal/failed-emails-digest.test.ts [hunk 3]
```diff
@@ -180,4 +158,240 @@ describe("with valid credentials", () => {
...
+    });
+    expect(updateEmailResponse.status).toBe(200);
+
+    // Send a test email that will fail
+    await niceBackendFetch("/api/v1/internal/send-test-email", {
+      method: "POST",
+      accessType: "admin",
+      headers: {
+        "x-stack-admin-access-token": adminAccessToken,
+      },
```
Comment (N2D4) [inline apps/e2e/tests/backend/endpoints/api/v1/internal/failed-emails-digest.test.ts]:
```suggestion
    // Remove primary email from the user
    const updateEmailResponse = await niceBackendFetch(`/api/v1/users/${userId}`, {
      method: "PATCH",
      accessType: "admin",
      body: {
        "primary_email": null,
      },
    });
    expect(updateEmailResponse.status).toBe(200);
    
    await Project.createAndSwitch({
      display_name: "Test Project No Owner Email",
    });

    // Send a test email that will fail
    await niceBackendFetch("/api/v1/internal/send-test-email", {
      method: "POST",
      accessType: "admin",
```

File: apps/e2e/tests/backend/endpoints/api/v1/internal/failed-emails-digest.test.ts [hunk 4]
```diff
@@ -180,4 +158,240 @@ describe("with valid credentials", () => {
...
+
+    const initialMessageCount = (await backendContext.value.mailbox.fetchMessages()).length;
+    const response = await niceBackendFetch("/api/v1/internal/failed-emails-digest", {
+      method: "POST",
+      headers: { "Authorization": "Bearer mock_cron_secret" }
+    });
+    expect(response.status).toBe(200);
+
+    const messages = await backendContext.value.mailbox.fetchMessages();
+    const newMessages = messages.slice(initialMessageCount);
```
Comment (N2D4) [inline apps/e2e/tests/backend/endpoints/api/v1/internal/failed-emails-digest.test.ts]: The tests will run in parallel, so this won't work. Instead, you'll have to filter by email (as you did in other tests)

Comment (N2D4) [reply apps/e2e/tests/backend/endpoints/api/v1/internal/failed-emails-digest.test.ts]: this also applies to everywhere else where you use .slice

**Review by N2D4 — CHANGES_REQUESTED** [review]
just some changes!
